tools/release.sh

```diff
@@ -184,27 +188,29 @@ versions=$(echo "$promotable" | cut -d: -f1)
 
 for version in $versions; do
   while true; do
-    files=$(echo "$promotable" | grep "^${version}" | sed 's/^'${version}': //')
-    echo -n "Promote ${version} files (${files})? [y/n] "
+    files=$(echo "$promotable" | grep "^${version}" | sed 's/^'"${version}"': //')
+    printf "Promote %s files (%s)? [y/n] " "${version}" "${files}"
     yorn=""
-    read yorn
+    read -r yorn
 
-    if [ "X${yorn}" == "Xn" ]; then
+    if [ "X${yorn}" = "Xn" ]; then
       break
     fi
 
     if [ "X${yorn}" != "Xy" ]; then
       continue
     fi
 
-		checktag $version
+		checktag "$version"
 
-    echo -e "\n# Promoting ${version}..."
+    echo ""
+    echo "# Promoting ${version}..."
 
-    ssh ${customsshkey} ${webuser}@${webhost} $promotecmd nodejs $version
+    # shellcheck disable=SC2029
+    ssh "${customsshkey}" ${webuser}@${webhost} $promotecmd nodejs "$version"
 
     if [ $? -eq 0 ];then
```

Comment:
This is weird code, the `ssh` command can just be put inside the `if` rather than run and then the $? checked.

```js
if ssh "${customsshkey}" ${webuser}@${webhost} $promotecmd nodejs "$version" ;then
```

Reply:
Also, you _probably_ want to partial-quote webuser/webhost and use the shorthand form